### DISCUSSION TITLE
File Uploader - Crop Configuration

### DISCUSSION THREAD

**Discussion by ciaran-deegan-davra** [uploadcare/file-uploader#391] [blocks]
Is it possible to configure custom crop ratios in the image uploader widget instead of just freeform cropping? I see it is documented for the mainline file uploader here: https://uploadcare.com/docs/uploads/file-uploader/#crop-config. Is there an equivalent config option for this new blocks file uploader?

**Comment by loookashow** [comment]
Hello, we plan to implement this feature in the near future and will notify you once it's completed.

**Comment by loookashow** [comment]
Created [the issue](https://github.com/uploadcare/blocks/issues/393)

**Reply by jacobdadalucs** [reply]
Do you have an estimated finishing time for this?

**Comment by martinsuhendra** [comment]
Hi @loookashow , do you have any idea when this very useful feature will come in?

**Comment by loookashow** [comment]
> Hi @loookashow , do you have any idea when this very useful feature will come in?

@martinsuhendra Yes, it's late September of '23. It is already on the go.

**Comment by nd0ut** [comment]
Hey @martinsuhendra,

We've released [v0.26.0](https://github.com/uploadcare/blocks/releases/tag/v0.26.0) with the `cropPreset` option. The documentation hasn't been deployed yet, so the links won't work. However, here is an example of how to use it: https://codesandbox.io/s/clever-silence-g8r8nr?file=/index.html

You can only define one crop preset at a time.

**Comment by martinsuhendra** [comment]
Hi, i really appreciate you informed me personally.

Really nice and clean feature when i tried it on sandbox, however i see the
limitation of one preset at a time. Would it be a future improvement to
have more than 1 preset in the future or this a permanent solution?

I use react environment to use this feature, does the component will change
reactively if the crop preset value/string is listening to react state?

Thank you so much!

Best Regards,
Martin Suhendra

**Reply by loookashow** [reply]
Hi @martinsuhendra, yes we're going to include this update to the future releases 🤩

**Reply by nd0ut** [reply]
Hey @martinsuhendra,

We've just released a fix (https://github.com/uploadcare/blocks/releases/tag/v0.27.3) that adds reactivity for the specified attributes. This means you can now switch crop presets on the fly.
